### PR TITLE
testing/docs: Include doctests in testing

### PR DESCRIPTION
**Pull request by carlpatt**
add a section to "tox.ini" that includes doctests in testing
add doctests to some modules

Changes proposed in this pull request:
I noticed that there were already doctests in some of the modules but they were not being tested. I Think it would be great to include them in testing with the added benefit of the doctests showing examples of how to use the modules. 

I got the idea to do this when I found the documentation bug months ago that was patched in [this commit](https://github.com/jmoiron/humanize/commit/15ecb0a778b38925bf8188bd1ca6be95b54ed1cc)

### CODE REVIEW

File: src/humanize/number.py [hunk 1]
```diff
@@ -17,6 +17,28 @@ def ordinal(value):
...
+        '103rd'
+        >>> ordinal(4)
+        '4th'
+        >>> ordinal(12)
+        '12th'
+        >>> ordinal(101)
+        '101st'
+        >>> ordinal(111)
+        '111th'
+        >>> ordinal('something else')
```
Comment (hugovk) [inline src/humanize/number.py:36]: Black style:
```suggestion
        >>> ordinal("something else")
```

File: src/humanize/number.py [hunk 2]
```diff
@@ -174,6 +243,7 @@ def fractional(value):
 
     Examples:
         ```pycon
+
```
Comment (hugovk) [inline src/humanize/number.py:246]:
```suggestion
```

File: src/humanize/time.py [hunk 3]
```diff
@@ -213,6 +213,10 @@ def naturalday(value, format="%b %d"):
     For date values that are tomorrow, today or yesterday compared to
     present day return representing string. Otherwise, return a string
     formatted according to `format`.
+
+    Examples:
```
Comment (hugovk) [inline src/humanize/time.py:217]: Missing examples?

File: src/humanize/filesize.py [hunk 4]
```diff
@@ -15,7 +15,20 @@ def naturalsize(value, binary=False, gnu=False, format="%.1f"):
...
+        >>> naturalsize(3000000)
+        '3.0 MB'
+        >>> naturalsize(300, False, True)
+        '300B'
+        >>> naturalsize(3000, False, True)
+        '2.9K'
+        >>> naturalsize(3000, False, True, "%.3f")
+        '2.930K'
+        >>> naturalsize(3000, True)
+        '2.9 KiB'
```
Comment (hugovk) [inline src/humanize/filesize.py:30]: Are these newlines needed?

File: src/humanize/filesize.py [hunk 5]
```diff
@@ -15,7 +15,20 @@ def naturalsize(value, binary=False, gnu=False, format="%.1f"):
     By default, decimal suffixes (kB, MB) are used.
 
     Non-GNU modes are compatible with jinja2's `filesizeformat` filter.
+    Examples:
```
Comment (hugovk) [inline src/humanize/filesize.py:18]: Please check the lint errors on the CI, newlines are needed before all "Examples" sections:

```suggestion

    Examples:
```

To run lint locally:

```sh
pip install -U tox
tox -e lint
```

Or:
```sh
pip install -U pre-commit
pre-commit install # Optional, to enable for commits
pre-commit run --all-files
```

**Review by hugovk — CHANGES_REQUESTED** [review]
Thanks for the PR!

**Comment by carlpatt** [comment]
@hugovk  I took care of all the whitespace issues

File: src/humanize/number.py [hunk 6]
```diff
@@ -17,6 +17,28 @@ def ordinal(value):
...
+        '12th'
+        >>> ordinal(101)
+        '101st'
+        >>> ordinal(111)
+        '111th'
+        >>> ordinal("something else")
+        'something else'
+        >>> ordinal(None) is None
+        True
+
```
Comment (hugovk) [inline src/humanize/number.py:40]: Also, is this newline needed, and those for subsequent ones?

Comment (carlpatt) [reply src/humanize/number.py:40]: Yes if there isn't a space between the three backticks and the test result doctests report it as a failure for some reason.

Comment (hugovk) [reply src/humanize/number.py:40]: Fair enough!

Yes, `pycon` is for syntax highlighting on RTD.

Thanks, I think this is ready for merge!

**Review by hugovk — APPROVED** [review]

**Pull request merged**